src/toil/batchSystems/singleMachine.py

```diff
@@ -52,6 +52,9 @@ def __init__(self, config, maxCores, maxMemory, maxDisk):
         AbstractBatchSystem.__init__(self, config, maxCores, maxMemory, maxDisk)
         assert self.maxCores >= self.minCores
         assert self.maxMemory >= 1
+
+        # MaxMemory will not be greater than the physically available memory on the machine
+        self.maxMemory = min(self.maxMemory, os.sysconf('SC_PAGE_SIZE') * os.sysconf('SC_PHYS_PAGES'))
```

Comment:
I'm going to push an improved version of this. Ask me about this after the next scrum.